### PR TITLE
fix: remove index in default maven/gradle single target extraction

### PR DESCRIPTION
**Pull request by danielmoy-google**
Previously we'd append index 0 to extract/preprocess steps for
maven/gradle, this makes a single-target extraction not have any index
at all, which is cleaner to read.

Split apart the setup not specific to a single target into a separate
preprocess step too, for clarity.

Also add a test for multi target extraction, to actually verify all of
this.

### CODE REVIEW

File: kythe/go/extractors/gcp/config/common.go [hunk 1]
```diff
@@ -64,11 +63,11 @@ func commonSteps(repo string) []*cloudbuild.BuildStep {
 	}
 }
 
-func preprocessorStep(build string, buildID int) *cloudbuild.BuildStep {
+func preprocessorStep(build string, targetID string) *cloudbuild.BuildStep {
```
Comment (jaysachs) [inline kythe/go/extractors/gcp/config/common.go]: s/targetID/idSuffix/ 

?

File: kythe/go/extractors/gcp/config/converter.go [hunk 2]
```diff
@@ -107,14 +108,23 @@ func KytheToBuild(conf *rpb.Config) (*cloudbuild.Build, error) {
...
+
+	build.Steps = append(build.Steps, g.preExtractSteps()...)
+
 	targets := hints.Targets
 	if len(targets) == 0 {
 		targets = append(targets, g.defaultExtractionTarget())
 	}
 	for i, target := range targets {
-		build.Steps = append(build.Steps, g.extractSteps(hints.Corpus, target, i)...)
+		id := ""
```
Comment (jaysachs) [inline kythe/go/extractors/gcp/config/converter.go]: idSuffix?

Comment (danielmoy-google) [reply kythe/go/extractors/gcp/config/converter.go]: ack

File: kythe/go/extractors/gcp/config/common.go (hunk 1 above)
Comment (danielmoy-google) [reply kythe/go/extractors/gcp/config/common.go]: sure we can do that

**Pull request merged**